src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/ClaimantResponseAgreedSettledPartAdmitDefendantLipNotificationHandler.java

```diff
@@ -49,33 +55,64 @@ public List<CaseEvent> handledEvents() {
 
     @Override
     public Map<String, String> addProperties(CaseData caseData) {
-        return Map.of(
-            RESPONDENT_NAME, getPartyNameBasedOnType(caseData.getRespondent1()),
-            CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference()
-        );
+        if (caseData.isRespondent1NotRepresented()) {
+            return Map.of(
+                RESPONDENT_NAME, getPartyNameBasedOnType(caseData.getRespondent1()),
+                CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference()
+            );
+        } else {
+            return Map.of(
+                CLAIM_REFERENCE_NUMBER, caseData.getLegacyCaseReference(),
+                CLAIM_LEGAL_ORG_NAME_SPEC, getRespondentLegalOrganizationName(caseData)
+            );
+        }
+    }
+
+    private String getRespondentLegalOrganizationName(CaseData caseData) {
+        Optional<Organisation> organisation = organisationService.findOrganisationById(caseData.getRespondent1OrganisationId());
```

Comment:
perhaps you could put the logic in OrganisationDetailsService? and then reuse it here?

Reply:
It's done